File: cpp/src/problem01/problem01.cpp

#include "problem01.h"

unsigned long long ProjectEuler::Problem01::solve() {
    unsigned long long sum = 0;
    for( int i = 0; i < limit; i++ ) {
        if( i % 3 == 0 || i % 5 == 0 ) {
            sum += i;
        }
    }
    return( sum );
}
/*
int main() {
    using namespace ProjectEuler;
    Problem01 problem(1000);
    std::cout << "The answer is: " << problem.solve() << std::endl;
    return 0;
}
*/

File: py/src/problems/problem01.py
from utils.projectEulerProblem import ProjectEulerProblem
class Problem01(ProjectEulerProblem):
    description = """If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6, and 9. The sum of these multiples is 23.
        
        Find the sum of all of the multiples of 3 or 5 below 1000"""
    limit = 1000
    
    def __init__(self):
        pass
    
    def solve(self) -> int:
        result = 0
        for i in range(self.limit):
            if i % 3 == 0 or i % 5 == 0:
                result += i

        return result